### PR TITLE
Migrate issue templates to issue forms

### PR DESCRIPTION
**Pull request by nscuro**
This brings a bit more structure into how issues are reported, and is a lot more user-friendly.

Also remove issue template for asking questions; Add various hints that issues should not be used to ask questions.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/defect-report.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      What version of Dependency-Track are you using?
+    options:
+    - 3.8.x
+    - 4.0.x
+    - 4.1.x
+    - 4.2.x
+    - 4.3.x
+    - 4.4.x
+    - 4.5.0
+    - 4.6.0-SNAPSHOT
```
Comment (msymons) [inline .github/ISSUE_TEMPLATE/defect-report.yml:70]: Needs to be updated,  now that 4.6.0 has been released

**Review by msymons — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/defect-report.yml (hunk 1 above)
Comment (nscuro) [reply .github/ISSUE_TEMPLATE/defect-report.yml]: Done!

**Comment by nscuro** [comment]
@stevespringett I'd like to get your sign off before I merge. Any objections?

FYI, you can preview the rendered issue forms here:
* https://github.com/DependencyTrack/dependency-track/blob/4c4722a91281301aca1f1e00f2678794070da702/.github/ISSUE_TEMPLATE/defect-report.yml
* https://github.com/DependencyTrack/dependency-track/blob/4c4722a91281301aca1f1e00f2678794070da702/.github/ISSUE_TEMPLATE/enhancement-request.yml

**Review by msymons — APPROVED** [review]
Looking good.  The version picker has been tweaked to include 4.7 SNAPSHOT (I have always feared logging issues for snapshots, not being sure if it was appropriate or not).

I understand that the "fluff" text gets removed from the logged issues.

**Comment by nscuro** [comment]
Thanks @msymons!

> I have always feared logging issues for snapshots, not being sure if it was appropriate or not

I think if we encourage users to test snapshots, we need a way to track defects they encounter.

**Comment by stevespringett** [comment]
This looks great @nscuro. Hopefully it will make our lives a bit easier and reduce the back and forth of asking for various things.

**Pull request merged**